### DISCUSSION TITLE
Image Extraction

### DISCUSSION THREAD

**Discussion by meghanaviyyapu** [pymupdf/PyMuPDF#3566] [Looking for help] [answered]
Could you let me know how an image can be created in the same shape as in the input PDF? After extracting an image from PDF, the image is in rectangular shape so during insertion, it is getting created in the same shape and not in the same shape as in the input PDF. Are there any attributes that affect the shape of an image?
For example, the image in the below PDF should be extracted and created in the same shape.
[OneWorld.pdf](https://github.com/user-attachments/files/15782862/OneWorld.pdf)

**Comment by JorjMcKie** [comment]
Sorry, no way.
All images are rectangular - always.
The visual effect (triangle) in your example is achieved outside / independently from the image, i.e. by drawing (vector graphics) a white triangle over it.

**Reply by JorjMcKie** [reply, marked as answer]
All you can do is **_not_** extracting the image, but instead making a pixmap using the image's bbox as clip.
This will again be a rectangular image, but with a white, triangular area ...

**Reply by meghanaviyyapu** [reply]
Thanks a lot for clarifying
